src/routes/character_routes.py

```diff
@@ -0,0 +1,79 @@
+from flask import render_template, request, redirect, url_for, session
+import logging
+from components import characters, users
+
+
+def character_routes(app):
+    
+
+    @app.route('/characters/list', methods=['GET'])
+    def my_characters_route():
+        error = None
+        user_id = session.get('user_id')
+
+        try:
+            character_list = characters.get_characters_by_user_id(user_id)
+        except Exception as e:
+                error = "Character list fetch failed."
+                logging.error(f"Error occurred: {str(e)}")
+
+        csrf_token = users.get_csrf_token()
+        return render_template('my_characters.html', character_list=character_list, error=error, csrf_token=csrf_token)
+
+
+    @app.route('/characters/create', methods=['GET', 'POST'])
+    def create_character_route():
```

Comment:
This function might benefit from separating the GET and POST into separate functions as mentioned in the last lecture. And also returning early when errors occur.

Reply:
Yeah. I didn't get to this yet as I had some trouble switching to Flask-Migrate, again because of lack of experience with postgre and Flask. I personally don't see a problem here but for the sake of the course I will look into it.